pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/timeseries/TimeSeriesPhysicalTableScan.java

```diff
@@ -54,6 +55,11 @@ public ExecutorService getExecutorService() {
     return _executorService;
   }
 
+  @Override
+  public BaseTimeSeriesPlanNode withInputs(List<BaseTimeSeriesPlanNode> newInputs) {
```

Comment:
What is the idea behind withInputs?

Reply:
This is to enable plan cloning, which allows replacing sub-trees of the plan.